### ISSUE TITLE
sirf_registration notebook example

### ISSUE THREAD

**Issue by DANAJK** [SyneRBI/SIRF-Exercises#59]
1.) When run in Jupyter from Docker, uses Python 2 and hence the `print` statements are ugly (though readable). 

2.) Clarity: Does the 1 in this statement `get_dimensions()[1]` refer to the first dimension? I thought python labeled the first dimension as 0? Oh, and python throws in another bit of confusion by putting the slice in this dimension. 
My point is that to anyone not versed in the quirks of python, this is quite hard to follow and needs some comments in the example.

3.) Does `GeometricalInfo` have a print method? This could be handy.

4.) The example might be made clearer by splitting into a few examples. First, two images that look similar apart from an in-plane displacement and then registering them. Second, two images that look like those in the example but with no real-world displacement (to show that the registration can take into account the spatial referencing provided by the geometrical info - the expected displacement field would be the identity). Third the full thing, but state at the outset how these images differ in their real-world position (at the moment it's not very clear if they are a fixed head with incorrect geometrical info, or there has been some movement between them that we are trying to find by registration).

**Comment by rijobro** [comment]
1. Sorry not sure how to help on that one. Is it possible to use python3?

2. Since it's a `NifttImageData`, the nifti convention is used here, in which the dimensions contain 7 values: `[ndim, nx, ny, nz, nt, nu, nv, nw]`. So here we used `get_dimensions()[1]` to get the number of voxels in the x direction. You're right that none of this is explained in the notebook and should probably be added. I suppose I omitted this in an attempt to not overcomplicating things.

3. I just had a look and it seems that `GeometricalInfo` has a method called `print_info`. It would be trivial to add a `__str__` method, which means you could do `print(ref.get_geometrical_info())`. I'm not sure why we didn't do this already.

P.S.: Confusingly, the GeometricalInfo would use `[nx,ny,nz]` which would mean that the 0th index would correspond to the 1st index of `NifttImageData.get_dimensions()`. I think it is somewhat unavoidable that the individually wrapped functionality (e.g., `NiftiImageData.get_dimensions`) are slightly different to native SIRF functionality (e.g., `GeometricalInfo`) since each package has its own way of ordering dimensions. STIR for example has the z dimension first.

**Comment by KrisThielemans** [comment]
@DANAJK is this issue relevant/fixable for 3.1?

**Comment by DANAJK** [comment]
I have made an alternative notebook that is on the Teams site.

**Comment by KrisThielemans** [comment]
all done I believe.
